modules/core/src/main/java/org/apache/ignite/internal/sql/SqlKeyword.java

```diff
@@ -347,7 +348,11 @@
      * @return {@code True} if it is a keyword.
      */
     public static boolean isKeyword(String str) {
-        return KEYWORDS.contains(str);
+        boolean isFallback = IgniteSystemProperties.getBoolean(IgniteSystemProperties.IGNITE_SQL_PARSER_DISABLE_H2_FALLBACK);
```

Comment:
It looks like we should skip check for keyword for some circumstances: for example in case of creating index we know that the next token is table name and we should not validate it. See`SqlCreateIndexCommand`:

`SqlQualifiedName tblQName = parseQualifiedIdentifier(lex);`

I think we should provide a special flag here that will skip the validation for table name.

Reply:
We should not skip checking `isKeyword` for most of cases. But only for specific ones

Reply:
ok